### PR TITLE
Replace GetTempFileName with GetRandomFileName

### PR DESCRIPTION
**Pull request by jbagga**
Addresses aspnet/Coherence-Signed#510

cc @ajaybhargavb @Eilon

### CODE REVIEW

File: src/NuGetPackageVerifier/Rules/AssemblyAttributesDataHelper.cs [hunk 1]
```diff
@@ -23,8 +23,12 @@ public static void SetAssemblyAttributesData(PackageAnalysisContext context)
                 if (extension.Equals(".dll", StringComparison.OrdinalIgnoreCase) ||
                     extension.Equals(".exe", StringComparison.OrdinalIgnoreCase))
                 {
-                    var assemblyPath = Path.ChangeExtension(
-                        Path.Combine(Path.GetTempPath(), Path.GetTempFileName()), extension);
+                    string assemblyPath;
+                    do
+                    {
+                        assemblyPath = Path.ChangeExtension(
+                        Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()), extension);
```
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/AssemblyAttributesDataHelper.cs]: Indent

File: src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs [hunk 2]
```diff
@@ -22,8 +22,12 @@ public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context
                 if (extension.Equals(".dll", StringComparison.OrdinalIgnoreCase) ||
                     extension.Equals(".exe", StringComparison.OrdinalIgnoreCase))
                 {
-                    var assemblyPath = Path.ChangeExtension(
-                        Path.Combine(Path.GetTempPath(), Path.GetTempFileName()), extension);
+                    string assemblyPath;
+                    do
+                    {
+                        assemblyPath = Path.ChangeExtension(
+                        Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()), extension);
```
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs]: Indent

File: src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs [hunk 3]
```diff
@@ -33,7 +37,7 @@ public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context
                     try
                     {
                         using (var packageFileStream = context.PackageReader.GetStream(currentFile))
-                        using (var fileStream = File.OpenWrite(assemblyPath))
+                        using (var fileStream = new FileStream(assemblyPath, FileMode.Create))
```
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs]: Why was this change necessary? `OpenWrite` creates the file if it doesn't exist.

Comment (jbagga) [inline src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs]: To keep it consistent with [AssemblyAttributesDataHelper](https://github.com/aspnet/BuildTools/blob/2ffa222f550f01451bc3439b28ae0b01507a73a1/src/NuGetPackageVerifier/Rules/AssemblyAttributesDataHelper.cs#L36), but I can undo it.

Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs]: Nah I was just curious.
